llvm/include/llvm/MCA/CustomBehaviour.h

```diff
@@ -123,13 +123,41 @@ class Instrument {
   /// The instrumentation data
   const StringRef Data;
 
+  std::optional<unsigned> Latency;
+
 public:
-  Instrument(StringRef Desc, StringRef Data) : Desc(Desc), Data(Data) {}
+  Instrument(StringRef Desc, StringRef Data) : Desc(Desc), Data(Data) {
+    // Skip spaces and tabs.
+    unsigned Position = Data.find_first_not_of(" \t");
+    if (Position >= Data.size())
+      // We reached the end of the comment. Bail out.
+      return;
+    Data = Data.drop_front(Position);
+    auto [Name, Value] = Data.split(':');
+    if (Name.upper() == "LATENCY") {
+      Position = Value.find_first_not_of(" \t");
+      if (Position >= Value.size())
+        return;
+      auto Stripped = Value.drop_front(Position);
+      unsigned L = 0;
+      if (!Stripped.getAsInteger(10, L))
+        Latency = L;
+    }
+  }
 
   Instrument() : Instrument("", "") {}
 
   virtual ~Instrument() = default;
 
+  virtual bool canCustomize() const { return bool(Latency); }
+  virtual void customize(InstrDesc &ID) const {
+    if (Latency) {
+      for (auto &W : ID.Writes)
+        W.Latency = *Latency;
```

Comment:
@adibiagio pointed this out in our offline discussions: an instruction may perform multiple writes, is there any way we could designate which write to apply this latency customization? It doesn't have to be in this patch, but a TODO comment would be nice.